projects/packages/stats/src/class-options.php

```diff
@@ -0,0 +1,149 @@
+<?php
+/**
+ * Stats Options
+ *
+ * @package automattic/jetpack-stats
+ */
+
+namespace Automattic\Jetpack\Stats;
+
+use Automattic\Jetpack\Constants;
+use Jetpack_Options;
+
+/**
+ * Stats Options class.
+ */
+class Options {
+
+	/**
+	 * Option name.
+	 *
+	 * @var string $option_name The 'stats' option name
+	 */
+	public static $option_name = 'stats_options';
+
+	/**
+	 * Stats Options.
+	 *
+	 * @var array $options An array of associated Stats options (default empty)
+	 */
+	protected static $options = array();
+
+	/**
+	 * Stats Get Options.
+	 *
+	 * @return array.
+	 */
+	public static function get_options() {
+		// Make sure we only get options from the database once per connection.
+		if ( count( self::$options ) > 0 ) {
+			return self::$options;
+		}
+
+		self::$options = get_option( self::$option_name, array() );
+
+		if ( ! isset( self::$options['version'] ) || self::$options['version'] < Constants::get_constant( 'STATS_VERSION' ) ) {
+			self::upgrade_options( self::$options );
```

Comment:
This could be a simpler method but I wanted to keep `upgrade_options` to ensure backwards compatibility.